### PR TITLE
[arp_update]: Use parent script PID for logger calls

### PR DESCRIPTION
**Pull request by theasianpianist**
#### Why I did it
When a script calls the logger utility, each individual call gets its own PID assigned. Rsyslogd creates an internal rate-limit object per PID that sends a log message. If the logger utility is called repeatedly, it will cause rsyslogd memory usage to increase permanently since a rate-limit object is being created for each call.

##### Work item tracking
- Microsoft ADO **28263484**:

#### How I did it
Override the `logger` command in the arp_update script to use the --id argument to specify the parent script PID that rsyslogd should use for rate limiting.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by prsunny** [comment]
Could you please update description and whats the behavior before and after this change?

**Review by prsunny — APPROVED** [review]

**Comment by prsunny** [comment]
@kperumalbfn , can you please review/merge?

**Review by kperumalbfn — APPROVED** [review]
LGTM

**Comment by prsunny** [comment]
@theasianpianist , can you confirm it is tested for the following scenarios and logging works as expected?
1. restart arp_update script 
2. restart rsyslogd

**Comment by theasianpianist** [comment]
> @theasianpianist , can you confirm it is tested for the following scenarios and logging works as expected?
> 1. restart arp_update script 
> 2. restart rsyslogd

Co firmed, logging works as expected here

File: files/scripts/arp_update [hunk 1]
```diff
@@ -11,7 +11,7 @@ ARP_UPDATE_VARS_FILE="/usr/share/sonic/templates/arp_update_vars.j2"
 
 # Overload `logger` command to include arp_update tag
 logger () {
-    command logger -t "arp_update" "$@"
+    command logger -i "$$" -t "arp_update" "$@"
```
Comment (qiluo-msft) [inline files/scripts/arp_update:14]:
```suggestion
    command logger --id="$$" -t "arp_update" "$@"
```

**Pull request merged**